pkg/manifests/external_dns.go

```diff
@@ -126,12 +127,21 @@ func (p Provider) string() string {
 	}
 }
 
+// InputExternalDNSConfig is the input configuration to generate ExternalDNSConfigs from the CRD or MC-level configuration
 type InputExternalDNSConfig struct {
 	TenantId, ClientId, InputServiceAccount, Namespace, InputResourceName string
-	Provider                                                              *Provider
-	IdentityType                                                          IdentityType
-	ResourceTypes                                                         map[ResourceType]struct{}
-	DnsZoneresourceIDs                                                    []string
+	// Provider is specified when an InputConfig is coming from the MC External DNS Reconciler, since no zones may be provided for the clean case
+	Provider *Provider
+	// IdentityType can either be MSI or WorkloadIdentity
+	IdentityType IdentityType
+	// ResourceTypes refer to the resource types that ExternalDNS should look for to configure DNS. These can include Gateway and/or Ingress
+	ResourceTypes map[ResourceType]struct{}
+	// DnsZoneresourceIDs contains the DNS zones that ExternalDNS will use to configure DNS
+	DnsZoneresourceIDs []string
+	// Filters contains various filters that ExternalDNS will use to filter resources it scans for DNS configuration
+	Filters *v1alpha1.ExternalDNSFilters
```

Comment:
I think this should be a new type instead of mixing crd and manifest domains.

Reply:
I was thinking about that -- 

The biggest reason why I chose to put it here is because all other forms of validation for input take place here, whether it's for input from the legacy scenario or from CRDs. That being said, we could just have that check earlier on and then marshall it into a separate intermediate type, but I don't think that we gain anything by doing that. If anything, since the filters are unique to the CRD case anyway, I figured we may as well keep that type propagated through as far as we could and then have a fixed domain that we're operating in to add these labels.

What do you think?